data_model/src/query/mod.rs

```diff
@@ -1352,7 +1348,7 @@ pub mod error {
             /// Parameter with id `{0}` not found
             Parameter(ParameterId),
             /// Failed to find public key: `{0}`
-            PublicKey(PublicKey),
+            PublicKey(Box<PublicKey>),
```

Comment:
I don't think it's recursion, but:

Before this change `PublicKey` stored a vec of bytes comprising the key, so its structure size wasn't that large. Now, however, it contains concrete key types with known sizes and so it stores the keys inline, leading to a larger structure size. 

I think it might even be enough to trigger clippy's [large_enum_variant](https://rust-lang.github.io/rust-clippy/master/index.html#/large_enum_variant) lint?

Reply:
Btw, I am not sure if it's a good idea to actually store they keys inline. It will definitely be a problem for using [zeroize](https://docs.rs/zeroize/latest/zeroize/) (which is needed for #3240), as it is not really possible to ensure a by-value struct gets deleted.

Reply:
you make a point with `zeroize`

Reply:
but `PublicKey` is not a secret and it doesn't have to be zeroed

Reply:
Yes, it's not recursion, it's large variant size. `PublicKey` takes a lot of space because of concrete crypto types which holds additional info